opentreemap/treemap/udf.py

```diff
@@ -358,17 +369,24 @@ def _validate_and_update_choice(
                 {'datatype': [_("Can't change choices "
                                 "on a non-choice field")]})
 
-        if old_choice_value not in datatype['choices']:
+        choices = datatype['choices']
+        protected_choices = datatype.get('protected_choices', [])
+        all_choices = choices + protected_choices
```

Comment:
Good idea. We'll still need a way to obtain `choices`, `protected_choices`, and `all_choices` for both UDFs and Collection UDFs (for `_validate_and_update_choice`). So should we have 6 functions in total to accomplish this?